pkg/apis/core/validation/cloudprofile.go

```diff
@@ -362,6 +366,16 @@ func validateContainerRuntimes(containerRuntimes []core.ContainerRuntime, fldPat
 	return allErrs
 }
 
+func validateArchitecture(arch *string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if arch != nil && *arch != v1beta1constants.Amd64 && *arch != v1beta1constants.Arm64 {
```

Comment:
We can enforce that architecture is `non-nil` here given that we default it to `amd64`.